aiida/tools/importexport/common/archive.py

```diff
@@ -311,37 +338,51 @@ def extract_tar(infile, folder, nodes_export_subfolder=None, silent=False):
     try:
         with tarfile.open(infile, 'r:*', format=tarfile.PAX_FORMAT) as handle:
 
-            try:
-                handle.extract(path=folder.abspath, member=handle.getmember('metadata.json'))
-            except KeyError:
-                raise CorruptArchive('required file `metadata.json` is not included')
+            if len(handle.getmembers()) == 1 and handle.getmembers()[0].size == 0:
+                raise CorruptArchive('no files detected in archive')
+
+            if silent:
```

Comment:
since you are repeating yourself here, could this code be moved outside?

Reply:
A lot of this I left as is, since I have another PR to completely redo all this code (#3242). Instead I have kept it as explicit as possible, but if you want I can try and minimize the code duplication here until #3242 will be implemented?